pkg/ls/compile.go

```diff
@@ -245,6 +248,65 @@ func (compiler *Compiler) compile(context *Context, ctx *compilerContext, ref st
 	return compiled, nil
 }
 
+func (compiler *Compiler) compileIncludeAttribute(context *Context, ctx *compilerContext) error {
+	nodeMap := make(map[*lpg.Node]*lpg.Node)
+	copySubtree := func(targetNode, includeNode *lpg.Node, tgtGraph *lpg.Graph, namespace string) error {
+		nodeMap[includeNode] = targetNode
+		srcTypes := lpg.NewStringSet(FilterAttributeTypes(targetNode.GetLabels().Slice())...)
+		includeTypes := FilterAttributeTypes(includeNode.GetLabels().Slice())
+		for _, typ := range includeTypes {
+			if !srcTypes.Has(typ) {
+				return ErrNotFound(fmt.Sprintf("attribute type for source: %v do not match with include: %v", targetNode, includeNode))
+			}
+			srcTypes.Add(typ)
+		}
+		for edges := includeNode.GetEdges(lpg.OutgoingEdge); edges.Next(); {
+			edge := edges.Edge()
+			targetNode.SetLabels(srcTypes)
+			if namespace != "" {
+				sfxIx := strings.LastIndex(namespace, "/")
+				if sfxIx != -1 {
+					SetAttributeID(targetNode, GetAttributeID(targetNode)+namespace[sfxIx:])
+				}
+			}
+			lpg.CopySubgraph(edge.GetTo(), tgtGraph, ClonePropertyValueFunc, nodeMap)
+			lpg.CopyEdge(edge, tgtGraph, ClonePropertyValueFunc, nodeMap)
+		}
+		return nil
+	}
+	context.GetLogger().Debug(map[string]interface{}{"mth": "compileIncludeAttribute"})
+	// Process until there are nodes with "include" attribute
+	for {
+		incNodes := compiler.CGraph.GetGraph().GetNodesWithProperty(IncludeSchemaTerm)
+		if !incNodes.Next() {
+			break
+		}
+		srcNode := incNodes.Node()
+		includeRef := AsPropertyValue(srcNode.GetProperty(IncludeSchemaTerm)).AsString()
+		srcNode.RemoveProperty(IncludeSchemaTerm)
+		includeSchema, err := compiler.loadSchema(ctx, includeRef)
+		if err != nil {
+			return err
+		}
+		if includeSchema == nil {
+			return ErrNotFound(includeRef)
+		}
+		includeRoot := includeSchema.GetSchemaRootNode()
+		if includeRoot == nil {
+			return ErrNotFound(includeRef)
+		}
```

Comment:
Here, check if FilterAttributeTypes(includeRoot.GetLabels() is the same as FilterAttributeTypes(srcNode.GetLabels())